armbalancer.go

```diff
@@ -72,16 +72,43 @@ func New(opts Options) http.RoundTripper {
 	}
 
 	if opts.TransportFactory == nil {
-		opts.TransportFactory = newRecyclableTransport
+		opts.TransportFactory = make(map[string]func(id int, parent *http.Transport, host string, port string, recycleThreshold, minReqsBeforeRecycle int64) http.RoundTripper)
```

Comment:
that func type signature is pretty long, let's make a type alias or interface for the sake of readability.